riscv/src/riscv.cpp

```diff
@@ -0,0 +1,54 @@
+#include "sim/memory.hpp"
+#include "sim/hart.hpp"
+#include "sim/simulator.hpp"
+#include "sim/instr.hpp"
+
+constexpr sim::PhysAddr DATA_SEGMENT_BASE_ADDR = 0x5000000000;
+constexpr size_t SIZE_16MB = size_t{1} << 24;
+constexpr size_t SIZE_2GB = size_t{1} << 31;
+namespace sim {
+
+enum RiscV {
```

Comment:
It will be better to crush with error message instead of error code return (because we are already in main)

Reply:
OK, fixed with cerr, 0/1/2 return error